docs/adr/007-configurability.md

```diff
@@ -0,0 +1,29 @@
+# ADR 007 - Configurability
+
+Date: 27-11-2024
+
+## Status
+
+Accepted
+
+Written years after the decision was made.
+
+## Context
+
+We know the system will be used in different contexts, from a simple setup with showing content on a couple of screens
+to large setups with hundreds of screens connected.
+
+Therefore, we want implementers to be able to configure the project to turn off features that will not be used.
+
+Environment variables can be used to configure a Symfony application.
+
+In the browser configuration can come from a file that is available to the client.
+
+## Decision
+
+We will make features configurable through environment and config files.
+
+## Consequences
+
+Features that are not "core" will have to be implemented as configurable wherever possible.
+This will introduce extra work when implementing features.
```

Comment:
Also one of the more negative consequences will be that you potentially have an amount of the application code as a "dead" part, maybe this will also have some negative influence on performance?

Reply:
I don't think there is a big negative impact on performance. But it will make the application code more complex, every time extra configuration is added.